### PR TITLE
Feature/create-stores

### PR DESCRIPTION
**Pull request by krishnaglodha**
Creating vector and raster store

### CODE REVIEW

File: src/geoserverx/_sync/gsx.py [hunk 1]
```diff
@@ -203,8 +240,9 @@ def get_allstyles(self) -> AllStylesModel:
             return results
 
     # Get specific style in GS
+
     @exception_handler
-    def get_style(self, style: str) -> StyleModel:
+    def get_style(self,  style: str) -> StyleModel:
```
Comment (francbartoli) [inline src/geoserverx/_sync/gsx.py]: Remove this useless change

File: src/geoserverx/_sync/gsx.py [hunk 2]
```diff
@@ -191,6 +205,29 @@ def get_raster_store(self, workspace: str, store: str) -> CoveragesStoreModel:
...
+        )
+        results = self.response_recognise(responses.status_code)
+        return results
+    
+    # delete store in specific workspaces
+    @exception_handler
+    def delete_store(self, workspace: str, store: str, type:str) ->  GSResponse: #TODO : add enum for type 
+        Client = self.http_client
+        if type == 'raster':
+            responses = Client.delete(f"/workspaces/{workspace}/coveragestores/{store}",headers=self.head)
```
Comment (francbartoli) [inline src/geoserverx/_sync/gsx.py]: spaces in the signature are not respected

File: src/geoserverx/_sync/gsx.py [hunk 3]
```diff
@@ -191,6 +205,29 @@ def get_raster_store(self, workspace: str, store: str) -> CoveragesStoreModel:
...
+        elif type == 'vector':
+            responses = Client.delete(f"/workspaces/{workspace}/datastores/{store}",headers=self.head)
```
Comment (francbartoli) [inline src/geoserverx/_sync/gsx.py]: Same above

File: src/geoserverx/_sync/gsx.py [hunk 4]
```diff
@@ -281,22 +319,78 @@ def get_all_layers(
...
         else:
             results = self.response_recognise(responses.status_code)
             return results
 
+    @exception_handler
+    def create_vector_layer(self, workspace: str, layer: FeatureTypesModel) -> GSResponse:
+        Client = self.http_client
+        responses = Client.post(
+            f"/workspaces/{workspace}/featuretypes",
+            data=json.dumps(layer.dict(by_alias=True,exclude_none=True)),
```
Comment (francbartoli) [inline src/geoserverx/_sync/gsx.py]: This is not pydantic v2 and also `json.dumps` should not be used

File: src/geoserverx/cli/cli.py [hunk 5]
```diff
@@ -384,7 +385,7 @@ def layer(
     """
     if request.value == "sync":
         client = SyncGeoServerX(username, password, url)
-        result = client.get_layer(workspace, layer).json()
+        result = client.get_layer(workspace, layer, detail).json()
```
Comment (francbartoli) [inline src/geoserverx/cli/cli.py]: this is not pydantic v2

File: src/geoserverx/models/coverages_layer.py [hunk 6]
```diff
@@ -0,0 +1,298 @@
+from enum import Enum
+from typing import Any, List, Optional,Union
+
+from pydantic import BaseModel, Field
+
+
+class _Key(Enum):
```
Comment (francbartoli) [inline src/geoserverx/models/coverages_layer.py]: Could you add docstring to understand what they represent?

File: src/geoserverx/models/coverages_layer.py [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+    regionateStrategy = 'regionateStrategy'
+    regionateFeatureLimit = 'regionateFeatureLimit'
+    cacheAgeMax = 'cacheAgeMax'
+    cachingEnabled = 'cachingEnabled'
+    regionateAttribute = 'regionateAttribute'
+    indexingEnabled = 'indexingEnabled'
+    dirName = 'dirName'
+
+class CRSDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/coverages_layer.py]: Why the class name refers to a dictionary?

File: src/geoserverx/models/coverages_layer.py [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+    shearX: Optional[float] = Field(None, description='shear value to apply in X')
+    shearY: Optional[float] = Field(None, description='shear value to apply in Y')
+    translateX: Optional[float] = Field(None, description='translation to apply in X')
+    translateY: Optional[float] = Field(None, description='translation to apply in Y')
+
+
+class InterpolationMethods(BaseModel):
+    string: Optional[List[str]] = None
+
+class supportedFormatsString(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/coverages_layer.py]: Class name should start capitalized

File: src/geoserverx/models/coverages_layer.py [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+    string: Optional[List[str]] = None
+
+class requestSRSString(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/coverages_layer.py]: Same above

File: src/geoserverx/models/coverages_layer.py [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+    string: Optional[List[str]] = None
+
+
+class interpolationMethodsString(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/coverages_layer.py]: Same above

File: src/geoserverx/models/featuretypes_layer.py [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+        None, description='A collection of data links for the resource.'
+    )
+class CRSetnry(BaseModel):
+    class Config:
+        allow_population_by_field_name = True
+    key: Optional[str] = Field(alias="@class")
+    dollar: Optional[str] = Field(..., alias="$")
+
+
+class CRSetnryDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/featuretypes_layer.py:50]: Typo in the class name that is referring surprisingly to a dictionary

File: src/geoserverx/models/featuretypes_layer.py [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+    )
+    length: Optional[int] 
+
+class Attributes(BaseModel):
+    attribute: Optional[List[AttributeItem]] = Field(
+        None, description='The derived set of attributes for the feature type.'
+    )
+
+
+class _Key(Enum):
```
Comment (francbartoli) [inline src/geoserverx/models/featuretypes_layer.py]: Is this duplicated?

**Review by francbartoli — CHANGES_REQUESTED** [review]

File: src/geoserverx/models/coverages_layer.py (hunk 7 above)
Comment (krishnaglodha) [reply src/geoserverx/models/coverages_layer.py:16]: Replaced the name with `NativeCRS`

File: src/geoserverx/models/coverages_layer.py (hunk 6 above)
Comment (krishnaglodha) [reply src/geoserverx/models/coverages_layer.py]: It is not needed, so I removed it

File: src/geoserverx/models/featuretypes_layer.py (hunk 12 above)
Comment (krishnaglodha) [reply src/geoserverx/models/featuretypes_layer.py]: Not needed

File: src/geoserverx/cli/cli.py (hunk 5 above)
Comment (krishnaglodha) [reply src/geoserverx/cli/cli.py]: fixed with model_dump_json

File: src/geoserverx/_sync/gsx.py (hunk 2 above)
Comment (krishnaglodha) [reply src/geoserverx/_sync/gsx.py]: Can you please explain what do you mean by this ?

File: src/geoserverx/_sync/gsx.py (hunk 4 above)
Comment (krishnaglodha) [reply src/geoserverx/_sync/gsx.py]: replaced with model_dump

File: src/geoserverx/_sync/gsx.py (hunk 3 above)
Comment (krishnaglodha) [reply src/geoserverx/_sync/gsx.py]: Can you please explain what do you mean by this ?

File: src/geoserverx/_sync/gsx.py (hunk 2 above)
Comment (francbartoli) [reply src/geoserverx/_sync/gsx.py]: there's no space after the comma, can we definitively enforce a ruff linter into another PR?

File: src/geoserverx/_sync/gsx.py (hunk 3 above)
Comment (francbartoli) [reply src/geoserverx/_sync/gsx.py]: Same answer above

**Comment by francbartoli** [comment]
There are conflicts to be resolved

**Comment by francbartoli** [comment]
I'm merging but the history becomes a mess in such a way

**Pull request merged**